### DISCUSSION TITLE
How to Detect and Handle Errors Only from refetchOnWindowFocus?

### DISCUSSION THREAD

**Discussion by DylanJu** [TanStack/query#7776] [Q&A] [answered]
I'm encountering an issue with refetchOnWindowFocus that I need help with.

In an iOS webview environment, when a query is refetched on window focus, a browser network error occasionally occurs. This happens because, at the OS level, the webview restricts network access if it has been in the background for a certain period of time. This behavior cannot be detected through browser APIs, and thus neither navigator.onLine nor addEventListener('online' | 'offline', callback) can catch this.

I would like to ignore such errors if cacheData is available. How can I handle this using useQuery?

**Comment by TkDodo** [comment, marked as answer]
we don't track what initiated the fetch, so I'm afraid there isn't much to cling onto.
